File: ai-hub-backend/src/models/promptModel.js
import mongoose from 'mongoose';

const historySchema = new mongoose.Schema({
  prompt: String,
  response: String,
  version: Number,
  createdAt: { type: Date, default: Date.now }
}, { _id: false });

const promptSchema = new mongoose.Schema({
  taskId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Task',
    required: true
  },
  name: { type: String, required: true },
  modelId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Model',
    required: true
  },
  promptText: { type: String, required: true },
  responseText: { type: String },
  history: [historySchema],
  executionOrder: {
    type: Number,
    required: true,
    default: 0
  },
  version: {
    type: Number,
    default: 1
  }
}, { timestamps: true });

export default mongoose.model('Prompt', promptSchema);

File: backend/infra/db/mongoose/models/prompt_model.go
package models

import "time"

type HistoryEntry struct {
	Prompt    string    `bson:"prompt"`
	Response  *string   `bson:"response,omitempty"`
	Version   int       `bson:"version"`
	CreatedAt time.Time `bson:"createdAt"`
}

type PromptDoc struct {
	ID             string         `bson:"_id"`
	TaskId         string         `bson:"taskId"`
	Name           string         `bson:"name"`
	ModelId        string         `bson:"modelId"`
	PromptText     string         `bson:"promptText"`
	ResponseText   *string        `bson:"responseText,omitempty"`
	History        []HistoryEntry `bson:"history"`
	ExecutionOrder int            `bson:"executionOrder"`
	Version        int            `bson:"version"`
	CreatedAt      time.Time      `bson:"createdAt"`
	UpdatedAt      time.Time      `bson:"updatedAt"`
}
